[12.0][MIG] bi_view_editor

astirpe opened the pull request:
This PR is the porting of `bi_view_editor` to V12 with the following improvements:

- [x] Store the JSON data structure in ORM
- [x] Improved robustness
- [x] Fix of fields/models translation
- [x] Add check for consistency of groups
- [x] Development of new tests
- [x] Add LEFT JOIN capabilities
- [x] Add sums and avg capabilities for tree views
- [x] Provide graph view for table relations

astirpe committed:
Initial commit bi_view_editor V9

astirpe committed:
[MIG+IMP] bi_view_editor: Migration to v10 + enhancements

* Add menu items creation feature
* Added selection of fields of a tree view
* Improved usability and strings made translatable
* Avoid display duplicated nodes
* Robustness
* Updated Dutch translation
* Avoid possible sql injection in bi_view_editor
* Removed deprecated RegistryManager

oca-transbot committed:
OCA Transbot updated translations from Transifex

tarteo committed:
[MIG] bi_view_editor: Migration to 11.0

SimoRubi committed:
[FIX] bi_view_editor: Apostrophe + migration script

Apostrophe in model name raised ValueError. Added needed migration script.

astirpe committed:
Remove logger warnings during update all

tarteo committed:
[FIX] Field widget, new readme + Make enabled in list view default true

astirpe committed:
Drop bi views when uninstalling module

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update bi_view_editor.pot

astirpe committed:
[12.0][MIG] bi_view_editor

MiquelRForgeFlow commented on bi_view_editor/models/ir_model.py:
```diff
@@ -0,0 +1,282 @@
+        return self.sudo().search(domain)
+
+    @api.model
+    def get_models(self, model_ids=None):
```
`model_ids={}` instead

astirpe replied:
This proposed change causes a warning:

```
************* Module bi_view_editor.models.ir_model
bi_view_editor/models/ir_model.py:173: [W0102(dangerous-default-value), IrModel.get_models] Dangerous default value {} as argument

```

so I will revert it back to `model_ids=None`.

MiquelRForgeFlow reviewed (approved):
code review :+1:

astirpe commented:
I'm adding tests but the **codecov/patch** is not increasing. The coverage is actually much higher, around 90%: https://codecov.io/gh/OCA/reporting-engine/tree/e99acf4ad56455afe6574307ac9712a3ed429cf6/bi_view_editor

astirpe commented:
@pedrobaeza do you know why the codecov/patch percentage does not increase?

astirpe committed:
Add extra functionalities

Add LEFT JOIN capabilities

Add sums and avg capabilities for tree views

Robustness and code review

Provide ER diagram view for table relations

patel1008b commented:
@astirpe @pedrobaeza @mreficent what do you think it will merge in the main branch. It's Badly required for one of project.

Thanks for this PR.

astirpe commented:
@bhaveshselarka this PR is not yet merged because of one extra review is still missing. You may want to test this PR on Runbot and report here the results. That would speed up the approval process. Thanks!

astirpe committed:
Add graphviz in .travis.yml

mreficent commented:
That monkey patch of the last commit is not obsolete, I think.

MiquelRForgeFlow commented:
I have a tested two migrations from v11. Without last commit, no problem; with last commit, it breaks. Please, remove last commit. That monkey patch of the last commit is not obsolete.

astirpe commented:
Thank you @mreficent !

patel1008b commented:
@astirpe @pedrobaeza @mreficent what do you think it will merge in the main branch. It's Badly required for one of project.

Thanks for this PR.

pedrobaeza commented:
@bhaveshselarka why don't you review the PR, try on runbot and give your feedback for speeding up the process?

patel1008b commented:
Yes, @pedrobaeza I think its best idea, Let me do that stuff.

I am at currently: https://runbot.odoo-community.org/runbot/repo/github-com-oca-reporting-engine-143

Where I can find a find this PR's correct instance. Can you please help me out in that?

pedrobaeza commented:
On the PR itself:
![Selección_008](https://user-images.githubusercontent.com/7165771/62764996-7bb41580-ba8f-11e9-8777-94ea7b0132e1.png)
I have launched a rebuild, as the instance has expired. Wait for the completion and you'll be able to access through the blue button. More details in https://odoo-community.org/page/review

HviorForgeFlow reviewed (approved):
LGTM

patel1008b commented:
@pedrobaeza @hveficent @astirpe @mreficent  will it merge in the main branch ?

astirpe commented:
@pedrobaeza the codecov is red but the real coverage is much higher, see also my https://github.com/OCA/reporting-engine/pull/293#issuecomment-501274907. There's something wrong with codecov, so for me the actual coverage is more than acceptable.

This PR has two approvals since a long time and I don't think we will get soon any other approvals. Is it okay to merge?

pedrobaeza commented:
Please squash a bit the commits and I'll proceed to merge. The codecov patch is base on previous commits done, so if you have done several pushes, it can get confused by that.

astirpe commented:
@pedrobaeza done!

astirpe commented:
@pedrobaeza would be great if we can get this PR merged. Thank you!

pedrobaeza commented:
/ocabot merge
